Src/FluentAssertions/Common/Guard.cs

```diff
@@ -63,6 +63,22 @@ public static void ThrowIfArgumentIsEmpty<T>(IEnumerable<T> values, string param
         }
     }
 
+    public static void ThrowIfArgumentIsEmpty(string str, string paramName, string message)
```

Comment:
Since these are public facing, they should be validating their arguments (validation inception lol).

In all seriousness, you don't want to see null reference exceptions being thrown by this guard method when `str` is `null` for example. That would become incredibly confusing. Same for other parameters.

Reply:
Fair enough.